clients/omnitruck/replicated/replicated.impl.go

```diff
@@ -0,0 +1,192 @@
+}
+
+type ReplicatedImpl struct {
+	ReplicatedConfig config.ReplicatedConfig
+	Client           HTTPClient
+	Logger           logger.Logger
+}
+
+func NewReplicatedImpl(config config.ReplicatedConfig, logger logger.Logger) IReplicated {
+	return &ReplicatedImpl{
+		Client:           &http.Client{},
+		ReplicatedConfig: config,
+		Logger:           logger,
+	}
+}
+
+func (r ReplicatedImpl) makeRequest(url, method, requestId string, payload io.Reader) (int, []byte, error) {
+	log := utils.AddLogFields("makeRequest", requestId, r.Logger)
+
+	req, err := http.NewRequest(method, url, payload)
+	if err != nil {
+		log.Errorln("error in creating new request.\n[ERROR] -", err.Error())
+		return 0, nil, err
+	}
+
+	req.Header.Add("accept", "application/json")
+	req.Header.Add("content-type", "application/json")
+	req.Header.Add("Authorization", r.ReplicatedConfig.Token)
+
+	res, err := r.Client.Do(req)
+	if err != nil {
+		log.Errorln("error on response.\n[ERROR] -", err.Error())
+		return 0, nil, err
+	}
+
+	defer res.Body.Close()
+	body, err := io.ReadAll(res.Body)
+	if err != nil {
+		log.Errorln("error on reading body.\n[ERROR] -", err.Error())
+		return 0, nil, err
+	}
+	return res.StatusCode, body, nil
+}
+
+func (r ReplicatedImpl) SearchCustomersByEmail(email string, requestId string) (customers []Customer, err error) {
+	log := utils.AddLogFields("SearchCustomersByEmail", requestId, r.Logger)
+
+	url := fmt.Sprintf("%s/customers/search", r.ReplicatedConfig.URL)
+	method := "POST"
+
+	payload := strings.NewReader(fmt.Sprintf(`{
+	  "app_id": "%s",
+	  "include_trial": true,
+	  "include_community": true,
+	  "include_paid": true,
+	  "include_dev": true,
+	  "include_active": true,
+	  "include_inactive": true,
+	  "query": "email:%s"
+  	}`, r.ReplicatedConfig.AppID, email))
+
+	respStatusCode, respBody, err := r.makeRequest(url, method, requestId, payload)
+	if err != nil {
+		log.Errorln("failed to search the customer: ", err.Error())
+		return nil, err
+	}
+
+	if respStatusCode != http.StatusOK {
+		err = fmt.Errorf("search customers by email failed with statusCode %d", respStatusCode)
+		log.Errorln("error on search customers by email.\n[ERROR] -", err.Error())
+		return nil, err
+	}
+
+	var respObj ResponseSearchCustomer
+	err = json.Unmarshal(respBody, &respObj)
+	if err != nil {
+		log.Errorln(constants.UNMARSHAL_ERR_MSG, err.Error())
+		return nil, err
+	}
+
+	return respObj.Customers, nil
+}
+
+func (r *ReplicatedImpl) PlatformVersionsAll(req *omnitruck.RequestParams, serverMode int) ([]omnitruck.ProductVersion, error) {
+	productVersions := []omnitruck.ProductVersion{}
+	flags := omnitruck.RequestParamsFlags{
+		Channel: true,
+	}
+	requestParams := omnitruck.ValidateRequest(req, flags)
+	if !requestParams.Ok {
+		r.Logger.Error("", requestParams.Message)
+		return productVersions, fiber.NewError(requestParams.Code, requestParams.Message)
+	}
+	if serverMode == 2 && req.Product == constants.PLATFORM_SERVICE {
```

Comment:
move this `2` to constant with proper name

Reply:
the serverMode is a int variable and 2 is a const for commercial that's why its 2